tests/integration/security/util/reachability/context.go

```diff
@@ -175,64 +107,92 @@ func (rc *Context) Run(testCases []TestCase) {
 			ctx.Logf("[%s] [%v] Finish waiting. Continue testing.", testName, time.Now())
 			ctx.Cleanup(func() {
 				if err := retry.UntilSuccess(func() error {
-					return rc.ctx.Config().DeleteYAML(c.Namespace.Name(), policyYAML)
+					return ctx.Config().DeleteYAML(c.Namespace.Name(), policyYAML)
 				}); err != nil {
 					log.Errorf("failed to delete configuration: %v", err)
 				}
 			})
-
-			for _, src := range []echo.Instance{rc.A, rc.B, rc.Headless, rc.Naked, rc.HeadlessNaked} {
-				for _, dest := range []echo.Instance{rc.A, rc.B, rc.Headless, rc.Multiversion, rc.Naked, rc.VM, rc.HeadlessNaked} {
-					copts := &callOptions
-					// If test case specified service call options, use that instead.
-					if c.CallOpts != nil {
-						copts = &c.CallOpts
+			for _, srcCluster := range ctx.Clusters() {
```

Comment:
Rather than looping over the clusters, you can loop over the echo.Instances themselves (they're slice-types) and then check the cluster using `echo.Config.Cluster`

https://github.com/istio/istio/blob/379eb242402dfec83f937c66e9dcfb38b8c5fe9b/tests/integration/pilot/common/routing.go#L383